test/integration/util.go

```diff
@@ -26,18 +26,17 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-func waitFor(ctx context.Context, interval time.Duration, check func(chan error)) error {
-	ch := make(chan error, 1)
-	go func() {
-		for {
-			check(ch)
+func waitFor(ctx context.Context, interval time.Duration, check func() (bool, error)) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			if done, err := check(); done {
```

Comment:
> @suskin re-requesting your review here as I have simplified the wait functionality quite a bit

kk. Looks nicer! :squid:

Originally I was wondering whether we needed both `done` and `err`, and it seems like it's useful to be able to return a `nil` err or a non-nil err on completion, so it does seem like we should have both. It may be worth adding a comment explaining why we have both though, so that when I forget and ask this again next time, the answer will already be there : )

Reply:
Great suggestion! Will update here :)

Reply:
Could you use `context.WithCancel` and the resutiling `cancel` function for part of this functionality?

Reply:
I had initially considered this when correcting the goroutine leak, but ended up going with the implementation you see here such that the writer of the `check` func would not have to be concerned with checking for the cancelled context. I could go either way on the implementation, do you have a strong feeling?

Reply:
I do not! Happy to merge.